anvil/src/eth/backend/mem/mod.rs

```diff
@@ -2247,7 +2247,7 @@ pub fn transaction_build(
 ) -> Transaction {
     let mut transaction: Transaction = eth_transaction.clone().into();
 
-    if let TypedTransaction::EIP1559(_) = eth_transaction.as_ref() {
+    if eth_transaction.is_dynamic_fee() {
```

Comment:
right, this properly "inverts" the check that used to happen below with `!is_eip1559`